src/taipy/core/config/core_section.py

```diff
@@ -93,6 +99,11 @@ def __init__(
         self.mode = mode or self._DEFAULT_MODE
         self.version_number = version_number or self._DEFAULT_VERSION_NUMBER
         self.force = force or self._DEFAULT_FORCE
+
+        self._core_version = core_version or self._DEFAULT_LATEST_CORE_VERSION
+        if self._core_version != self._DEFAULT_LATEST_CORE_VERSION:
+            raise ConfigCoreVersionMismatched(self._core_version, self._DEFAULT_LATEST_CORE_VERSION)
```

Comment:
We don't want to check compatibility here but in the update method.
```suggestion
        self._core_version = core_version
```

Reply:
I think we can expand the check no? in both update and init as well, does it not make sense?

Reply:
it will also help keep a value for core_service when serialize